src/main/java/net/minecraftforge/common/ForgeConfig.java

```diff
@@ -104,6 +107,12 @@
                     .worldRestart()
                     .defineInRange("clumpingThreshold", 64, 64, 1024);
 
+            multipartHandler = builder
+                    .comment("Determines the handler for all multipart behavior in the instance. If empty (default), will be populated with the first mod-supplied handler available, or default to the vanilla one if none are present.")
```

Comment:
Surely relying on the first mod handler is a race condition between multiple handlers given modloading is both threaded and without consistent ordering? Perhaps crashing to force a particular one to be selected in the event of multiple justifies having a config option more?

Reply:
I agree that picking one completely at random isn't the best idea ever, but forcing a crash on players who are quickly putting together a pack with a handful of mods they like just because there are multiple multipart handlers to choose from doesn't sounds like the best user experience ever either. I would argue that showing a popup that lets you pick which one is going to render most people as clueless as they'd be if they had to edit the config.

My thought process was that, in the case of multiple being present and none being specified in the config, one (essentially at random) would be picked, and that'd be what'd be used from there onwards. Implementations should really not vary that much when it comes to behavior, but there is always the option to pick one manually via the config file if the specific need for it arises.

Reply:
With that last commit I just made, I un-derped the system as it was in the first commit. I didn't quite get the new config system so I kind of botched it, and it just clicked in my head a second ago. Multipart handlers are now properly linked to server config loading, unlike before, which could've been rather problematic when switching game saves on the client.